src/Control/FunFlow/Base.hs

```diff
@@ -43,24 +91,36 @@ runNoEffect = error "Impossible!"
 
 type SimpleFlow = Flow NoEffect SomeException
 
-step :: Store b => (a -> IO b) -> Flow eff ex a b
-step = effect . Step
+step :: (a -> b) -> Flow eff ex a b
+step = step' def
+
+step' :: Properties a b -> (a -> b) -> Flow eff ex a b
+step' props = effect . Step props
+
+stepIO :: (a -> IO b) -> Flow eff ex a b
+stepIO = stepIO' def
 
-named :: Store b => T.Text -> (a -> b) -> Flow eff ex a b
-named n f = effect $ Named n f
+stepIO' :: Properties a b -> (a -> IO b) -> Flow eff ex a b
+stepIO' props = effect . StepIO props
 
-external :: ContentHashable a => (a -> ExternalTask) -> Flow eff ex a CS.Item
+named :: T.Text -> (a -> b) -> Flow eff ex a b
+named n = step' (def { name = Just n})
+
+external ::  ContentHashable IO a => (a -> ExternalTask) -> Flow eff ex a CS.Item
 external = effect . External
 
 wrap :: eff a b -> Flow eff ex a b
-wrap = effect . Wrapped
+wrap = effect . Wrapped def
+
+wrap' :: Properties a b -> eff a b -> Flow eff ex a b
+wrap' p eff = effect $ Wrapped p eff
 
-docker :: ContentHashable a => (a -> Docker.Config) -> Flow eff ex a CS.Item
+docker ::  ContentHashable IO a => (a -> Docker.Config) -> Flow eff ex a CS.Item
```

Comment:
Why two whitespaces before the constraint?
Same below

Reply:
Bad search/replace!